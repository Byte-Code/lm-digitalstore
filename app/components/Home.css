.container {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  padding: 32px 0;
  height: 100vh;
}

.container > img {
  width: 60%;
}

.container > h1 {
  margin-top: 32px;
  font-weight: lighter;
}
