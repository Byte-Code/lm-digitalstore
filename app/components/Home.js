import React, { Component } from 'react';
import styles from './Home.css';


export default class Home extends Component {
  render() {
    return (
      <div>
        <div className={styles.container}>
          <img src="https://upload.wikimedia.org/wikipedia/commons/d/d4/Leroy_Merlin.svg" alt="logo" />
          <h1>Digital Store</h1>
        </div>
      </div>
    );
  }
}
