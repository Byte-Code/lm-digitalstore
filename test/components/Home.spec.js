import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import mountWithStore from '../../app/utils/testingUtils';
import Home from '../../app/components/Home';
import World from '../../mocks/world';

// eslint-disable-next-line arrow-body-style
jest.mock('../../app/components/DynamicTemplates', () => {
  return {
    Comfort: () => <div>1</div>
  };
});


describe('Home', () => {
  const requestFetchWorld = jest.fn();

  it('should render null if world is empty', () => {
    const emptyWorld = fromJS({});
    const result = shallow(
      <Home
        world={emptyWorld}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render properly if world is not empty', () => {
    const result = shallow(
      <Home
        world={fromJS(World)}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should call requestFetchWorld after mount', () => {
    mountWithStore(
      <Home
        world={fromJS(World)}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(requestFetchWorld).toHaveBeenCalled();
  });
});
