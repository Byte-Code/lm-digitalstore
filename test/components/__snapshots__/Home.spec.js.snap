// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render null if world is empty 1`] = `null`;

exports[`Home should render properly if world is not empty 1`] = `
<div>
  1
</div>
`;
